plugins/backup/nas/src/main/java/org/apache/cloudstack/backup/NASBackupProvider.java

```diff
@@ -0,0 +1,442 @@
+        } catch (AgentUnavailableException e) {
+            throw new CloudRuntimeException("Unable to contact backend control plane to initiate backup");
+        } catch (OperationTimedoutException e) {
+            throw new CloudRuntimeException("Operation to initiate backup timed out, please try again");
+        }
+
+        if (answer != null && answer.getResult()) {
+            return backupDao.remove(backup.getId());
+        }
+
+        return false;
+    }
+
+    @Override
+    public Map<VirtualMachine, Backup.Metric> getBackupMetrics(Long zoneId, List<VirtualMachine> vms) {
+        final Map<VirtualMachine, Backup.Metric> metrics = new HashMap<>();
+        if (CollectionUtils.isEmpty(vms)) {
+            LOG.warn("Unable to get VM Backup Metrics because the list of VMs is empty.");
+            return metrics;
+        }
+
+        for (final VirtualMachine vm : vms) {
+            Long vmBackupSize = 0L;
+            Long vmBackupProtectedSize = 0L;
+            for (final Backup backup: backupDao.listByVmId(null, vm.getId())) {
+                vmBackupSize += backup.getSize();
+                vmBackupProtectedSize += backup.getProtectedSize();
+            }
+            Backup.Metric vmBackupMetric = new Backup.Metric(vmBackupSize,vmBackupProtectedSize);
+            LOG.debug(String.format("Metrics for VM [uuid: %s, name: %s] is [backup size: %s, data size: %s].", vm.getUuid(),
+                    vm.getInstanceName(), vmBackupMetric.getBackupSize(), vmBackupMetric.getDataSize()));
+            metrics.put(vm, vmBackupMetric);
+        }
+        return metrics;
+    }
+
+    @Override
+    public boolean assignVMToBackupOffering(VirtualMachine vm, BackupOffering backupOffering) {
+        return Hypervisor.HypervisorType.KVM.equals(vm.getHypervisorType());
+    }
+
+    @Override
+    public boolean removeVMFromBackupOffering(VirtualMachine vm) {
+        return true;
+    }
+
+    @Override
+    public boolean willDeleteBackupsOnOfferingRemoval() {
+        return false;
+    }
+
+    @Override
+    public void syncBackups(VirtualMachine vm, Backup.Metric metric) {
+        // TODO: check and sum/return backups metrics on per VM basis
+    }
+
+    @Override
+    public List<BackupOffering> listBackupOfferings(Long zoneId) {
+        final List<BackupRepository> repositories = backupRepositoryDao.listByZoneAndProvider(zoneId, getName());
+        final List<BackupOffering> offerings = new ArrayList<>();
+        for (final BackupRepository repository : repositories) {
+            offerings.add(new NasBackupOffering(repository.getName(), repository.getUuid()));
+        }
+        return offerings;
+    }
+
+    @Override
+    public boolean isValidProviderOffering(Long zoneId, String uuid) {
+        return true;
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey[]{
+        };
+    }
+
+    @Override
+    public String getName() {
+        return "nas";
```

Comment:
use string constant, also use the same in NasBackupOffering.java , etc

Reply:
All other providers export string name, constants aren't required because of this. We can check maybe introduce them like enums if necessary. (same pattern used across several other types of plugins)